src/core/data/ndarray/ndarray_vector.h

```diff
@@ -193,15 +261,175 @@ class NdArrayVector
         return const_cast<DataType&>(static_cast<NdArrayVector const&>(*this)(indexes));
     }
 
-    static const std::size_t dimension = dim;
-    using type                         = DataType;
+
+    auto shape() const { return nCells_; }
+
+    template<typename Mask>
+    auto operator[](Mask&& mask)
+    {
+        return MaskedView{*this, std::forward<Mask>(mask)};
+    }
+
 
 private:
     std::array<std::uint32_t, dim> nCells_;
     std::vector<DataType> data_;
 };
 
 
+class NdArrayMask
+{
+public:
+    NdArrayMask(std::size_t min, std::size_t max)
+        : min_{min}
+        , max_{max}
+    {
+    }
+
+    NdArrayMask(std::size_t width)
+        : min_{width}
+        , max_{width}
+    {
+    }
+
+    template<typename Array>
+    void fill(Array& array, typename Array::type val) const
+    {
+        if constexpr (Array::dimension == 1)
+            fill1D(array, val);
+
+        else if constexpr (Array::dimension == 2)
+            fill2D(array, val);
+
+        else if constexpr (Array::dimension == 3)
+            fill3D(array, val);
+    }
+
+    template<typename Array>
+    void fill1D(Array& array, typename Array::type val) const
+    {
+        auto shape = array.shape();
+
+        for (std::size_t i = min_; i <= max_; ++i)
+            array(i) = val;
+
+        for (std::size_t i = shape[0] - 1 - max_; i <= shape[0] - 1 - min_; ++i)
+            array(i) = val;
+    }
+
+    template<typename Array>
+    void fill2D(Array& array, typename Array::type val) const
+    {
+        auto shape = array.shape();
+
+        // left border
+        for (std::size_t i = min_; i <= max_; ++i)
+            for (std::size_t j = min_; j <= shape[1] - 1 - max_; ++j)
+                array(i, j) = val;
+
+        // right border
+        for (std::size_t i = shape[0] - 1 - max_; i <= shape[0] - 1 - min_; ++i)
+            for (std::size_t j = min_; j <= shape[1] - 1 - max_; ++j)
+                array(i, j) = val;
+
+
+        for (std::size_t i = min_; i <= shape[0] - 1 - min_; ++i)
+        {
+            // bottom border
+            for (std::size_t j = min_; j <= max_; ++j)
+                array(i, j) = val;
+
+            // top border
+            for (std::size_t j = shape[1] - 1 - max_; j <= shape[1] - 1 - min_; ++j)
+                array(i, j) = val;
+        }
+    }
+
+    template<typename Array>
+    void fill3D(Array& array, typename Array::type val) const
+    {
+        throw std::runtime_error("3d not implemented");
+    }
+
+
+    auto min() const { return min_; };
+    auto max() const { return max_; };
+
+private:
+    std::size_t min_, max_;
+};
+
+
+
+
+template<typename Array, typename Mask>
+void operator>>(MaskedView<Array, Mask>&& inner, MaskedView<Array, Mask>&& outer)
+{
+    using MasekdView_t = MaskedView<Array, Mask>;
+
+    if constexpr (MasekdView_t::dimension == 1)
```

Comment:
Masked not Masekd (my bad)
same in the second and third constexpr and above using

Reply:
good catch!